pkg/server/forward.go

```diff
@@ -51,68 +53,74 @@ func init() {
 	prometheus.MustRegister(overwrittenTimestamps)
 }
 
-type Store struct {
-	next   store.Store
-	url    *url.URL
-	client *http.Client
-	logger log.Logger
-}
+// ForwardHandler gets a request containing metric families and
+// converts it to a remote write request forwarding it to the upstream at fowardURL.
+func ForwardHandler(logger log.Logger, forwardURL *url.URL) http.HandlerFunc {
+	client := http.Client{}
 
-func New(logger log.Logger, url *url.URL, next store.Store) *Store {
-	return &Store{
-		next:   next,
-		url:    url,
-		client: &http.Client{},
-		logger: log.With(logger, "component", "store/forward"),
-	}
-}
+	return func(w http.ResponseWriter, r *http.Request) {
+		partitionKey, ok := PartitionFromContext(r.Context())
```

Comment:
let's rename this `partitionKey` thing. It's a relic from the `partitioned` metrics type but it's a  weird name IMO. I think something like `ID` or anything would be much more clear

Reply:
Yeah, I had the same thought during refactoring. I would have gone for `clusterID`. That's what it's ultimately is, right?

Reply:
yes exactly